src/Files.App/Views/Settings/TagsPage.xaml.cs

```diff
@@ -87,14 +92,22 @@ private void RemoveTag_Click(object sender, RoutedEventArgs e)
 		private void RenameTextBox_TextChanged(object sender, TextChangedEventArgs e)
 		{
 			var text = ((TextBox)sender).Text;
-			editingTag!.IsNameValid = IsNameValid(text);
+			editingTag!.IsNameValid = IsNameValid(text) || (text == editingTag!.Tag.Name);
+			editingTag!.CanCommit = IsNameValid(text) && ((text != editingTag!.Tag.Name) || (editingTag!.NewColor != editingTag!.Tag.Color));
+		}
+
+		private void EditColorPicker_ColorChanged(ColorPicker sender, ColorChangedEventArgs args)
+		{
+			if (editingTag is null) return;
+
+			editingTag!.CanCommit = IsNameValid(editingTag!.NewName) && (editingTag!.NewName != editingTag!.Tag.Name || (CommunityToolkit.WinUI.Helpers.ColorHelper.ToHex(sender.Color) != editingTag!.Tag.Color));
```

Comment:
I think it would be more readable it it were splitted into multiple lines

```suggestion
			editingTag!.CanCommit = IsNameValid(editingTag!.NewName) &&
			    (editingTag!.NewName != editingTag!.Tag.Name || (CommunityToolkit.WinUI.Helpers.ColorHelper.ToHex(sender.Color) != editingTag!.Tag.Color));
```

Reply:
I've done this in a slightly different way, please take a look

Reply:
Yes that's fine